be/src/vec/functions/array/function_array_remove.h

```diff
@@ -0,0 +1,363 @@
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#pragma once
+
+#include "vec/columns/column_array.h"
+#include "vec/columns/column_const.h"
+#include "vec/data_types/data_type_array.h"
+#include "vec/data_types/data_type_number.h"
+#include "vec/functions/function.h"
+#include "vec/functions/function_helpers.h"
+
+namespace doris::vectorized {
+
+class FunctionArrayRemove : public IFunction {
+public:
+    static constexpr auto name = "array_remove";
+    static FunctionPtr create() { return std::make_shared<FunctionArrayRemove>(); }
+
+    /// Get function name.
+    String get_name() const override { return name; }
+
+    bool use_default_implementation_for_nulls() const override { return true; }
+
+    bool is_variadic() const override { return false; }
+
+    size_t get_number_of_arguments() const override { return 2; }
+
+    DataTypePtr get_return_type_impl(const DataTypes& arguments) const override {
+        DCHECK(is_array(arguments[0]))
+                << "First argument for function: " << name << " should be DataTypeArray but it has type "
+                << arguments[0]->get_name() << ".";
+        return arguments[0];
+    }
+
+    Status execute_impl(FunctionContext* context, Block& block, const ColumnNumbers& arguments,
+                        size_t result, size_t input_rows_count) override {
+
+
+//        auto dst_null_column = ColumnUInt8::create(input_rows_count);
+//        UInt8* dst_null_map = dst_null_column->get_data().data();
+//        const UInt8* src_null_map = nullptr;
+//        ColumnsWithTypeAndName args;
+//        auto col_left = block.get_by_position(arguments[0]);
+//        if (col_left.column->is_nullable()) {
+//            auto null_col = check_and_get_column<ColumnNullable>(*col_left.column);
+//            src_null_map = null_col->get_null_map_column().get_data().data();
+//            args = {{null_col->get_nested_column_ptr(), remove_nullable(col_left.type),
+//                     col_left.name},
+//                    block.get_by_position(arguments[1])};
+//        } else {
+//            args = {col_left, block.get_by_position(arguments[1])};
+//        }
+//        auto res_column = _execute_non_nullable(args, input_rows_count, src_null_map);
+
+        // For default implementation of nulls args
+        ColumnsWithTypeAndName args = {
+                block.get_by_position(arguments[0]), block.get_by_position(arguments[1])
+        };
+
+        LOG(INFO) << "left type is " << args[0].type->get_name();
+        LOG(INFO) << "right type is " << args[1].type->get_name();
+        LOG(INFO) << "right column type is " << args[1].column->get_name();
```

Comment:
Logging in IO path should be simplified.

Reply:
done